torch/_functorch/aot_autograd.py

```diff
@@ -1858,22 +1858,39 @@ def backward(ctx, *flat_args):
                 list(ctx.symints) + list(ctx.saved_tensors) + list(contiguous_args)
             )
             del contiguous_args
-            if CompiledFunction.compiled_bw is None:
-                # TODO - pass in fake tensors ?
-                context = disable_autocast_manager if disable_amp else nullcontext
-                with context(), track_graph_compiling(aot_config, "backward"):
-                    CompiledFunction.compiled_bw = aot_config.bw_compiler(
-                        bw_module, all_args
+
+            class CompiledFunctionBackward(torch.autograd.Function):
+                # This custom autograd Function ensures that the backward graph is
+                # properly connected, but errors when the user performs double backward.
+                #
+                # See comment for why once_differentiable is not sufficient:
+                # https://github.com/pytorch/pytorch/pull/92348/files#r1072962107
+                @staticmethod
+                def forward(ctx, *all_args):
+                    all_args_list = list(all_args)
```

Comment:
hmmm.... don't we need to clear this list 🤔 (@eellison @soulitzer)

Reply:
oh actually maybe not, we're calling `call_func_with_args(all_args_list, steal_args=True)` later

Reply:
I think you might be right... we need to `del all_args` after `all_args_list = list(all_args)`. `all_args_list` will probably be cleared by `call_func_with_args`, but all_args still holds a reference to the tensor.